src/System.CommandLine/Help/HelpBuilder.cs

```diff
@@ -624,18 +722,31 @@ private bool ShouldDisplayArgumentHelp(ICommand? command)
             return command.Arguments.Any(ShouldShowHelp);
         }
 
-        private int GetConsoleWindowWidth()
+        private int GetConsoleWindowWidth(IConsole console)
         {
             try
             {
-                return System.Console.WindowWidth;
+                if (console is SystemConsole && System.Console.WindowWidth > 0)
```

Comment:
The thing to watch out for here is that `System.Console.WindowWidth` will throw there's no window (e.g., output is redirected). We generally avoid calling `System.Console` directly for this reason. You can't get this information unless there's actually a terminal (the `ITerminal` abstraction in `System.CommandLine.Rendering`.

I think we can probably away with the exception handler here altogether and do something like this instead:

```cs
private static int GetConsoleWindowWidth(SystemConsole console)
{
  return console.IsOutputRedirected
    ? int.MaxValue
    : System.Console.WindowWidth;
}
```

It might make sense to move this to the `SystemConsole` class in fact.

Reply:
I can move it.